install.sh

```diff
@@ -491,6 +577,70 @@ build_from_tarball_icu()
     pop_directory
 }
 
+build_from_tarball_libpng()
+{
+    local URL=$1
+    local ARCHIVE=$2
+    local REPO=$3
+    local JOBS=$4
+    shift 4
+
+    if [[ !($BUILD_LIBPNG) ]]; then
+        initialize_libpng_packages
+        display_message "LIBPNG build not enabled"
+        return
+    fi
+
+    display_message "Download $ARCHIVE"
+
+    create_directory $REPO
+    push_directory $REPO
+
```

Comment:
Missing a push/pop here and below - to avoid extracting into own directory.

We are voilating DRY here. Now that we have identical tar builds we should generalize this function.